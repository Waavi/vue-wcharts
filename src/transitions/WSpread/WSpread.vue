<script>
import VueTypes from 'vue-types'
import WTrans from '../WTrans/WTrans.vue'

export default {
    name: 'WSpread',
    functional: true,
    props: {
        axis: VueTypes.oneOf(['x', 'y']).isRequired,
        transition: VueTypes.string.isRequired,
    },

    render (h, { children, props, parent }) {
        // eslint-disable-next-line no-underscore-dangle
        const id = `WSpread${parent._uid}`
        const { axis, transition } = props

        return h('g', [
            h('defs', [
                h(
                    'clipPath',
                    {
                        attrs: {
                            id,
                        },
                    },
                    [
                        h(
                            WTrans,
                            {
                                props: {
                                    initialProps: {
                                        width: axis === 'x' ? 0 : '100%',
                                        height: axis === 'y' ? 0 : '100%',
                                    },
                                    transition,
                                },
                            },
                            [
                                h('rect', {
                                    attrs: {
                                        x: 0,
                                        y: 0,
                                        width: '100%',
                                        height: '100%',
                                    },
                                }),
                            ]
                        ),
                    ]
                ),
            ]),
            h(
                'g',
                {
                    attrs: {
                        'clip-path': `url(#${id})`,
                    },
                },
                children
            ),
        ])
    },
}
</script>
