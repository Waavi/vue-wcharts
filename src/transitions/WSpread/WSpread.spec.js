import { mount, shallowMount } from '@vue/test-utils'
import WSpread from './WSpread.vue'

describe('Widgets/WSpread', () => {
    const propsData = {
        id: 'test',
        axis: 'x',
        transition: 'all 250s 250ss ease',
    }

    const defaultConfig = {
        context: {
            props: propsData,
        },
    }

    xit(`Should be render correctly`, () => {
        const wrapper = shallowMount(WSpread, defaultConfig)
        expect(wrapper.html()).toMatchSnapshot()
    })

    xit(`Should be render correctly with children and axis it is x`, () => {
        const wrapper = mount({
            components: { WSpread },
            props: {
                axis: 'x',
                transition: 'all 250s 250ss ease',
            },
            template: `
              <div>
                <WSpread :transition="transition" axis="x">
                    <span>Hello</span>
                </WSpread>
              </div>
            `,
        })
        expect(wrapper.html()).toMatchSnapshot()
    })

    xit(`Should be render correctly with children and axis it is y`, () => {
        const wrapper = mount({
            components: { WSpread },
            props: {
                axis: 'y',
                transition: 'all 250s 250ss ease',
            },
            template: `
              <div>
                <WSpread :transition="transition" axis="y">
                    <span>Hello</span>
                </WSpread>
              </div>
            `,
        })
        expect(wrapper.html()).toMatchSnapshot()
    })
})
