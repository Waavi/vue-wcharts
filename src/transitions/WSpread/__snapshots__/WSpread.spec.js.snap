// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widgets/WSpread Should be render correctly 1`] = `
<g>
  <defs>
    <clipPath id="WSpread0">
      <rect x="0" y="0" width="100%" height="100%"></rect>
    </clipPath>
  </defs>
  <g clip-path="url(#WSpread0)"></g>
</g>
`;

exports[`Widgets/WSpread Should be render correctly with children and axis it is x 1`] = `
<div>
  <g>
    <defs>
      <clipPath id="WSpread3">
        <rect x="0" y="0" width="100%" height="100%"></rect>
      </clipPath>
    </defs>
    <g clip-path="url(#WSpread3)"><span>Hello</span></g>
  </g>
</div>
`;

exports[`Widgets/WSpread Should be render correctly with children and axis it is y 1`] = `
<div>
  <g>
    <defs>
      <clipPath id="WSpread6">
        <rect x="0" y="0" width="100%" height="100%"></rect>
      </clipPath>
    </defs>
    <g clip-path="url(#WSpread6)"><span>Hello</span></g>
  </g>
</div>
`;
